File: src/main/java/com/pCarpet/controller/ReportController.java
package com.pCarpet.controller;

import com.pCarpet.model.Report;
import com.pCarpet.model.User;
import com.pCarpet.services.ReportService;
import com.pCarpet.services.UserService;

import java.sql.Timestamp;
import java.util.List;

import javax.servlet.http.HttpServletRequest;

import org.springframework.ui.Model;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

@Controller
@RequestMapping("/Report")
public class ReportController {

	private ReportService reportService;
	private UserService userService;

	@Autowired
	public ReportController(ReportService reportService , UserService userService)  {
		this.reportService = reportService;
		this.userService = userService;
	}
	
	
	@RequestMapping(value = "/driver/reports", method = RequestMethod.GET) 
	public String driverReports(HttpServletRequest request, Model model ) {
		return "reportDriver";
	}
	
	@RequestMapping(value = "/driver/hystory", method = RequestMethod.GET) 
	public String driverHystory(HttpServletRequest request, Model model ) {
		
		User user = userService.getLoggedUser();
		List<Report> reports = reportService.getAllReportModels(user);
		model.addAttribute("reports", reports);		
		return "reportHystory";
	}
	
	
	@RequestMapping(value = "/driver/addReport", method = RequestMethod.POST) 
	public String driverAddReport(HttpServletRequest request, Model model ) {
		
		User user = userService.getLoggedUser();
		Timestamp time = new Timestamp(System.currentTimeMillis());
		int type = Integer.parseInt(request.getParameter("type").toString());
		String description = request.getParameter("description").toString();		
		
		Report report = new Report(0,type, description, time.toString(), "Non gestita" ,user);
		reportService.insertReport(report);

		return "homeDriver";
	}
	
	@RequestMapping(value = "/driver/ownerReport", method = RequestMethod.GET) 
	public String driverOwnerReport(HttpServletRequest request, Model model ) {
		String usernameOwner = "gestore"; //da sistemare
		User user = new User();
		user.setUsername(usernameOwner);
		List<Report> reportOwner = reportService.getAllReportModels(user);
		model.addAttribute("reports", reportOwner);
		return "reportHystory";
	}	
	
	@RequestMapping(value = "/owner/reportUser", method = RequestMethod.GET) 
	public String ownerReportuser(HttpServletRequest request, Model model ) {
		List<Report> reports = reportService.getAllReportOwner();
		model.addAttribute("reports", reports);
		return "reportOwner";
	}
	
	@RequestMapping(value = "/owner/addReport", method = RequestMethod.GET) 
	public String ownerAddReport(HttpServletRequest request, Model model ) {
		return "addReportOwner";
	}
	
	@RequestMapping(value = "/owner/addedReport", method = RequestMethod.POST) 
	public String ownerAddedReport(HttpServletRequest request, Model model ) {
		User user = userService.getLoggedUser();
		String description = request.getParameter("description").toString();
		int type = 0; //owner type
		Timestamp time = new Timestamp(System.currentTimeMillis());
		
		Report report = new Report(0,type, description, time.toString(), "In lavorazione" ,user);
		reportService.insertReport(report);
		return "homeOwner";
	}
	
	@RequestMapping(value = "/owner/hystory", method = RequestMethod.GET) 
	public String ownerHystory(HttpServletRequest request, Model model ) {
		User user = userService.getLoggedUser();
		List<Report> reportsOwner = reportService.getAllReportModels(user);
		model.addAttribute("reports", reportsOwner);
		return "reportHystory";
	}
	
	@RequestMapping(value = "/back", method = RequestMethod.GET) 
	public String back(HttpServletRequest request, Model model ) {
		User user = userService.getLoggedUser();
		if(user.getType().equalsIgnoreCase("driver")) {return "homeDriver";}
		else { return "addReportOwner";}
		
	}		
	
}

File: PCarpet/Controllers/ReportController.cs
﻿using PCarpet.Service;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

using System.Text;
using PCarpet.DTO;

namespace PCarpet.Controllers
{
    public class ReportController : Controller
    {
        // REPORTS:

        // DA DRIVER:
        // - non gestito = 0;

        // DA OWNER:
        // - appena inviato = 3;


        UserService userService;
        ReportService reportService;
        private double latitude;
        private double longitude;

        public ReportController()
        {
            userService = new UserService();
            reportService = new ReportService();
        }

        //GET
        public ActionResult driverReports()
        {
            return View("reportDriver");
        }

        //GET
    public ActionResult driverHystory()
        {
            user user = userService.getLoggedUser();
            List<ReportDTO> reports = reportService.getAllReportDTO(user.username, latitude, longitude);

            ViewBag.reports = reports;
            return View("reportHystory");
        }

        //POST
    public ActionResult driverAddReport(String description, String type)
        {
            user user = userService.getLoggedUser();
            DateTime time = DateTime.Now;

            String mydescription = description;
            int mytype = 0;
            if (int.TryParse(type, out mytype))
            {
                //non gestita
                reportService.insertReport( new ReportDTO(mytype, mydescription, time, user.username, null, latitude, longitude), null );
            }
            return View("homeDriver");
        }

        //GET
    public ActionResult driverOwnerReport()
        {
            String usernameOwner = "gestore"; //da sistemare
            user user = new user();
            user.username = usernameOwner;
            List<ReportDTO> reportOwner = reportService.getAllReportDTO(user.username, latitude, longitude);
            ViewBag.reports = reportOwner;
            return View("reportHystory");
        }

            //GET
    public ActionResult ownerReportuser()
        {
            List<ReportDTO> reports = reportService.getAllReportOwner(latitude, longitude);
            ViewBag.reports = reports;
            return View("reportOwner");
        }

        //GET
    public ActionResult ownerAddReport()
        {
            return View("addReportOwner");
        }

    //POST
    public ActionResult ownerAddedReport(String description)
        {
            user user = userService.getLoggedUser();
            String mydescription = description;

            DateTime time = DateTime.Now;

            // lo stato della segnalazione 3 è lo stato iniziale della segnalazione inviata da owner
            int type = user.type; //owner type

            reportService.insertReport(new ReportDTO(type, mydescription, time, user.username, null, latitude, longitude), null);

            return View("homeOwner");
        }

        //GET
    public ActionResult ownerHystory()
        {
            user user = userService.getLoggedUser();
            List<ReportDTO> reportsOwner = reportService.getAllReportDTO(user.username,latitude,longitude);

            ViewBag.reports = reportsOwner;
            return View("reportHystory");
        }

        //GET
    public ActionResult back()
        {
            user user = userService.getLoggedUser();
            //type = 1 è driver, type != è owner
            if (user.type == 1) { return View("homeDriver"); }
            else { return View("addReportOwner"); }
        }
    }
}
